xs2a-adapter-service-api-adapter/src/main/java/de/adorsys/xs2a/adapter/http/ApacheHttpClient.java

```diff
@@ -120,12 +130,49 @@ public String content(Request.Builder requestBuilder) {
             InputStream content = entity != null ? entity.getContent() : EmptyInputStream.INSTANCE;
 
             T responseBody = responseHandler.apply(statusCode, content, responseHeaders);
+            logResponse(response, responseHeadersMap, responseBody);
             return new Response<>(statusCode, responseBody, responseHeaders);
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
 
+    private <T> void logResponse(
+        CloseableHttpResponse response,
+        Map<String, String> headers,
+        T responseBody
+    ) {
+        logger.debug("<-- {}", response.getStatusLine());
+        headers.keySet().forEach(
+            key -> logger.debug("<-- {}: {}", key, headers.get(key))
+        );
+        logger.debug("<--");
+        Optional<String> contentType = Optional.ofNullable(
+            headers.get(ResponseHeaders.CONTENT_TYPE)
+        );
+        if (contentType.isPresent() && contentType.get().startsWith("application/json")) {
+            try {
+                logger.debug(RESPONSE_BODY, new ObjectMapper().writeValueAsString(responseBody));
```

Comment:
Why do you serialize just deserialized object instead of logging it before response handler?

Reply:
removed at all because of new requirements

Reply:
What new requirements?

Reply:
https://jira.adorsys.de/browse/XS2AAD-115